gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -299,7 +304,7 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
 
         // Depending on if DagManager is present, handle execution
         if (this.dagManager.isPresent()) {
-          submitFlowToDagManager((FlowSpec) spec, jobExecutionPlanDag);
+          submitFlowToDagManager(flowSpec, jobExecutionPlanDag);
```

Comment:
let's consider dropping support for running GaaS w/o the DagManager.  what are your thoughts @Will-Lo ?

(not as part of this PR, but as a follow-up in the next month or so)

Reply:
Yeah I think we shouldn't need to support a scenario where dagmanager is optional given how important it is for lifecycle management